### ISSUE TITLE
Build failure: tts

### ISSUE THREAD

**Issue by ak-1** [NixOS/nixpkgs#263362]
### Steps To Reproduce

1. build tts: nix-shell -p tts

### Build log

```
=========================== short test summary info ============================
FAILED tests/pytest_tests/unit_tests/test_lib/test_redir.py::test_tqdm_progbar[Redirect] - ValueError: signal only works in main thread of the main interpreter
FAILED tests/pytest_tests/unit_tests/test_lib/test_redir.py::test_reinstall[Redirect] - ValueError: signal only works in main thread of the main interpreter
= 2 failed, 755 passed, 16 skipped, 5 xfailed, 4 xpassed, 161 warnings in 57.84s =
/nix/store/37p8gq9zijbw6pj3lpi1ckqiv18j2g62-stdenv-linux/setup: line 1594: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/333fgxi6z8xhq7p6saqkpgwpi1zwdgyx-python3.10-wandb-0.15.2.drv' failed with exit code 1;
       last 10 log lines:
       > tests/pytest_tests/unit_tests_old/tests_launch/test_launch.py:1079: 16 warnings
       >   /build/source/tests/pytest_tests/unit_tests_old/tests_launch/test_launch.py:1079: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
       >     @pytest.mark.timeout(320)
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED tests/pytest_tests/unit_tests/test_lib/test_redir.py::test_tqdm_progbar[Redirect] - ValueError: signal only works in main thread of the main interpreter
       > FAILED tests/pytest_tests/unit_tests/test_lib/test_redir.py::test_reinstall[Redirect] - ValueError: signal only works in main thread of the main interpreter
       > = 2 failed, 755 passed, 16 skipped, 5 xfailed, 4 xpassed, 161 warnings in 57.84s =
       > /nix/store/37p8gq9zijbw6pj3lpi1ckqiv18j2g62-stdenv-linux/setup: line 1594: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/333fgxi6z8xhq7p6saqkpgwpi1zwdgyx-python3.10-wandb-0.15.2.drv'.
error: 1 dependencies of derivation '/nix/store/1qvvqp0w6r9wmqjfsa2qwgfriplpnnfc-python3.10-k-diffusion-0.0.14.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6r77myyw7dlq22y02vv8d68j60fak9rk-tts-0.14.0.drv' failed to build
```

### Additional context

This is on latest NixOS 23.05 on x86_64 GNU/Linux and AMD GPU.

### Notify maintainers

@mweinelt
@Mic92 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.0-rc5, NixOS, 23.05 (Stoat), 23.05.20231022.b3ddf96`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.6`
 - channels(root): `"home-manager, nixos-23.05"`
 - channels(ak): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by mweinelt** [comment]
cc @samuela for wandb.

**Comment by samuela** [comment]
Hmm I'm not seeing any build wandb failures on either x86_64-linux nor aarch64-darwin as of 0e18a0a0b4092f3f52bacfca3c0db8f7c66702d2:
```
❯ nix-build -A python3Packages.wandb
/nix/store/daz6x90y9x0bigy365pg8g0kg2pi7ilm-python3.11-wandb-0.15.11

❯ git log -1
commit 0e18a0a0b4092f3f52bacfca3c0db8f7c66702d2 (HEAD -> master, upstream/master, upstream/HEAD)
```

**Comment by mweinelt** [comment]
The report is for nixos-23.05.

**Comment by ak-1** [comment]
I can confirm that this issue does not exists under nixos-unstable.

**Comment by Mic92** [comment]
Sorry. I currently don't have the capacity to fix this in the 23.05 nixos release. However since 23.11 is around the corner, feel free to cherry pick from there.
